codec_specs.md

```diff
@@ -1,21 +1,61 @@
 
-*   CodecPrivate in the TrackEntry
-*   CodecState in the BlockGroup
+The intention behind this list is not to list all existing audio and video codecs, but rather to list those codecs that are currently supported in Matroska and therefore need a well defined `Codec ID` so that all developers supporting Matroska will use the same `Codec ID`. If you feel we missed support for a very important codec, please tell us on our development mailing list (cellar at ietf.org).
 
-Each of these elements contain the same kind of data. And these data depend on the codec used.
+## Defining Matroska Codec Support
 
-Important Note:
+Support for a codec is defined in Matroska with the following values.
 
-The intention behind this list is `NOT` to list all existing audio and video codecs, but rather to list those codecs that are `currently supported` in Matroska and therefore need a well defined codec ID so that all developers supporting Matroska will use the same ID. If you feel we missed support for a very important codec, please tell us on our development mailing list (matroska-devel at lists.matroska.org).
+### Codec ID
 
+Each codec supported for storage in Matroska MUST have a unique `Codec ID`. Each `Codec ID` MUST be prefixed with the string from the following table according to the associated type of the codec. All characters of a `Codec ID Prefix` MUST be capital letters (A-Z) except for the last character of a `Codec ID Prefix` which MUST be a hypen ("-").
+
+Codec Type | Codec ID Prefix
+-----------|----------------
+Video      | "V_"
+Audio      | "A_"
+Subtitle   | "S_"
+Button     | "B_"
+
+Each `Codec ID` MUST include a `Major Codec ID` immediately following the `Codec ID Prefix`. A `Major Codec ID` MAY be followed by an OPTIONAL `Codec ID Suffix` to communicate a refinement of the `Major Codec ID`. If a `Codec ID Suffix` is used, then the `Codec ID` MUST include a forward slash ("/") as a separator between the `Major Codec ID` and the `Codec ID Suffix`. The `Major Codec ID` MUST be composed of only capital letters (A-Z) and numbers (0-9). The `Codec ID Suffix` MUST be composed of only capital letters (A-Z), numbers (0-9), and formard slash ("/").
+
+The following table provides examples of valid `Codec IDs` and their components:
+
+Codec ID Prefix | Major Codec ID | Separator | Codec ID Suffix | Codec ID
+A_              | AAC            | /         | MPEG2/LC/SBR    | A_AAC/MPEG2/LC/SBR
+V_              | V_MPEG4        | /         | ISO/ASP         | V_MPEG4/ISO/ASP
+V_              | MPEG1          |           |                 | V_MPEG1
+
+### Codec Name
+
+Each encoding supported for storage in Matroska MUST have a Codec Name. The Codec Name provides a readable label for the encoding.
+
+### Description
+
+An optional description for the encoding.
```

Comment:
Maybe we should clarify that `Codec Name` and `Description` are meant solely for human consumption and have no effect on the processing of Matroska data by a demuxer.